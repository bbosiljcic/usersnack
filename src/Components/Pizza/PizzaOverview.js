import React from 'react';

export default function PizzaOverview() {
  return (
    <h1>PizzaOverview</h1>
  )
}
