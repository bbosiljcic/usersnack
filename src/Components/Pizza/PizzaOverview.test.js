import React from 'react';
import renderer from 'react-test-renderer';

import pizzaService from '../../services/pizza';
import PizzaOverview from './PizzaOverview';

it('renders correctly when there are no pizzas', () => {
  const tree = renderer.create(<PizzaOverview />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a single pizza correctly', () => {
  const fakePizza = {
    id: 1,
    img: test.jpg,
    incredients: ['ing1', 'ing2', 'ing3'],
    name: 'Test Pizza',
    price: 100,
  };

  jest.spyOn(pizzaService, 'getAll').mockImplementation(() => [fakePizza]);

  const tree = renderer.create(<PizzaOverview />).toJSON();
  expect(tree).toMatchSnapshot();
});
