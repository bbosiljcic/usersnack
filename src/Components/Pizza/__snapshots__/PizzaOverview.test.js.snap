// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single pizza correctly 1`] = `
<div>
  <h2>
    Get one of our freshly made pizzas
  </h2>
  <div
    className="MuiPaper-root MuiCard-root pizza-item MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="pizza-item--content"
    >
      <div>
        <h3>
          Test Pizza
        </h3>
        <span
          className="pizza-item--incredients"
        >
          ing1, ing2 and ing3
        </span>
      </div>
      <div
        className="pizza-item--actions"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Order a 
            Test Pizza
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="pizza-item--price"
        >
          $ 
          100.00
        </div>
      </div>
    </div>
    <img
      alt="tasty Test Pizza"
      src="/assets/pizzas/undefined"
    />
  </div>
</div>
`;

exports[`renders correctly when there are no pizzas 1`] = `
<div>
  <h2>
    Get one of our freshly made pizzas
  </h2>
</div>
`;
